framework/include/materials/FunctorMaterial.h

```diff
@@ -0,0 +1,45 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Material.h"
+
+/**
+ * FunctorMaterials compute functor material properties
+ */
+class FunctorMaterial : public Material
+{
+public:
+  static InputParameters validParams();
+
+  FunctorMaterial(const InputParameters & parameters);
+  void computeProperties() override final {}
```

Comment:
Should this API maybe just go straight into the Material class rather than in a child class?

Reply:
If it's not its own class then it's impossible to avoid triplicate addition of `FunctorMaterials` like we do for regular materials. And then I can't do any error checking to see whether a functor property has multiple definitions because I store them all in `SubProblem`